tee-worker/ts-tests/identity.test.ts

```diff
@@ -12,24 +12,24 @@ import {
     assertInitialIDGraphCreated,
     checkUserShieldingKeys,
 } from './common/utils';
-
+import { env_network } from './common/helpers';
 import { hexToU8a, u8aConcat, u8aToHex, u8aToU8a, stringToU8a } from '@polkadot/util';
 import { step } from 'mocha-steps';
 import { assert } from 'chai';
-import { IdentityGenericEvent, TransactionSubmit } from './common/type-definitions';
-import { HexString } from '@polkadot/util/types';
 import { multiAccountTxSender, sendTxsWithUtility } from './common/transactions';
 import { assertIdentityVerified, assertIdentityCreated, assertIdentityRemoved } from './common/utils';
 import type { LitentryPrimitivesIdentity } from '@polkadot/types/lookup';
 import type { LitentryValidationData } from './parachain-interfaces/identity/types';
+import type { TransactionSubmit } from './common/type-definitions';
+import type { HexString } from '@polkadot/util/types';
 import { Event } from '@polkadot/types/interfaces';
 import { ethers } from 'ethers';
-const substrateExtensionIdentity: LitentryPrimitivesIdentity = {
+const substrateExtensionIdentity = {
     Substrate: {
         address: '0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48', //Bob
         network: 'Litentry',
     },
-} as any;
+} as unknown as LitentryPrimitivesIdentity;
```

Comment:
I get why we need the `as` when interfacing with the (mis-)augmented API, but why the assertion here?

If I get it correctly this is the _input_ we want to pass to `createType('LitentryPrimitivesIdentity', <_>)`, but from what we saw earlier _that_ type is actually different from how `LitentryPrimitivesIdentity` is declared, no? For instance, this input doesn't have the `type` or `isSubstrate` props.

This might still be a job for the `ExtractEnumVariant` we discussed earlier :thinking:

Reply:
>This might still be a job for the ExtractEnumVariant we discussed earlier

Your understanding is correct. If we want perfection, we do need to do it that way(ExtractEnumVariant). However, you will find that in our ts-test, we mostly use interfaces for `LitentryPrimitivesIdentity`. I'm thinking that when assembling the parameters, we can define the `LitentryPrimitivesIdentity` type ,but need  pass it to `createType` with ` Substrate: {
        address: '0x8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48', 
        network: 'Litentry',
    }.` Do you think this is significant? Would it be enough for IDHub to know the  `LitentryPrimitivesIdentity` which interfaces it has?

If they want to define `LitentryPrimitivesIdentity`, would it be enough for them to assemble their own `ExtractEnumVariant`?

Reply:
We have decided to address this matter in another PR/issue.